src/main/java/teammates/ui/webapi/action/RegenerateStudentCourseLinksAction.java

```diff
@@ -0,0 +1,73 @@
+package teammates.ui.webapi.action;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+import teammates.common.util.EmailSendingStatus;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.StringHelper;
+import teammates.common.util.Templates;
+import teammates.ui.webapi.output.RegenerateStudentCourseLinksData;
+
+/**
+ * Regenerates the course join and feedback session links for a given student in a course.
+ */
+public class RegenerateStudentCourseLinksAction extends Action {
+
+    private static final String SUCCESSFUL_UPDATE = "Student's links for this course have been regenerated,";
+    private static final String SUCCESSFUL_UPDATE_WITH_EMAIL = SUCCESSFUL_UPDATE + " and the email has been sent.";
+    private static final String SUCCESSFUL_UPDATE_BUT_EMAIL_FAILED = SUCCESSFUL_UPDATE + " but the email failed to send.";
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!Config.isDevServer()) {
+            throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmailAddress = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+        StudentAttributes student = logic.getStudentForEmail(courseId, studentEmailAddress);
+        if (student == null) {
+            return new JsonResult(new RegenerateStudentCourseLinksData("The student with the email " + studentEmailAddress
```

Comment:
Should be 404 here?

Reply:
Yup this should 404 based on similar convention with other API.